includes/collection/class-outbox.php

```diff
@@ -278,4 +288,51 @@ public static function maybe_get_activity( $outbox_item ) {
 
 		return self::get_activity( $outbox_item );
 	}
+
+	/**
+	 * Get the object ID of an activity.
+	 *
+	 * @param Activity $activity The activity object.
+	 * @return string The object ID.
+	 */
+	private static function get_object_id( $activity ) {
+		// Most common.
+		if ( is_object( $activity->get_object() ) ) {
+			return $activity->get_object()->get_id();
+		}
+
+		// Rare.
+		if ( is_string( $activity->get_object() ) ) {
+			return $activity->get_object();
+		}
+
+		// Exceptional.
+		return $activity->get_actor() ?? $activity->get_id();
```

Comment:
wouldn't this return the generated ID with the hash when the actor is empty?

Reply:
Yes, if it ever got to that. I'm not sure it will?

Reply:
this is a waste of data and I would even recommend this generation of a default ID from the code!

Reply:
Follow-up PR?

Reply:
the default ID was only because we had no outbox activities and it was required to have a unique id by remote services.

Reply:
but then, this snippet would return `null` in that case and store an empty id and we would again deal with it! I think it is important to have this discussion before merging this!

Reply:
Sure! What would you suggest as a more appropriate fallback to `actor` in Activities without `object`?

Reply:
I'm not sure we even need a fallback for it? [Reviewing the spec](https://www.w3.org/TR/activitystreams-vocabulary/#activity-types), it looks like only `Question` doesn't have either an `object` or `actor` field.

Reply:
We could use `name` in that case?

Reply:
It is not about if an `Activity` supports it, it is about if it provides it. And because it is not required it might fail or return something unexpected.

The `name` is no valid ID and might cause issues if we might use the meta as ID in the future.